### PR TITLE
fix(firebase_core): Fix `FirebaseException` error code bug by making default value: "unknown".

### PR DESCRIPTION
**Pull request by hyungtaecf**
## Description

Fixing "Unhandled Exception: type 'Null' is not a subtype of type 'String'". 
Field code was receiving null from event['error']['code']. Now if it receives null it becomes the default value "unknown".

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by russellwheatley** [comment]
@hyungtaecf, could you please sign the CLA and we will look to get this fix reviewed & merged. Thanks.

**Comment by hyungtaecf** [comment]
I have signed CLA right away, how can I update it here?

**Comment by russellwheatley** [comment]
@hyungtaecf, simply copy: `@googlebot I signed it!`, paste it in the comment box & press `comment`.

**Review by Salakar — APPROVED** [review]

**Comment by hyungtaecf** [comment]
@googlebot I signed it!

File: packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_exception.dart [hunk 1]
```diff
@@ -30,9 +30,9 @@ class FirebaseException implements Exception {
   FirebaseException({
     required this.plugin,
     this.message,
-    this.code = 'unknown',
+    String? code,
     this.stackTrace,
-  });
+  }), this.code = code ?? 'unknown';
```
Comment (russellwheatley) [inline packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_exception.dart:35]:
```suggestion
  }): this.code = code ?? 'unknown';
```
You need to use a colon for invoking an initializer list.

**Review by russellwheatley — CHANGES_REQUESTED** [review]
update this and we can get this merged.

File: packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_exception.dart (hunk 1 above)
Comment (russellwheatley) [reply packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_exception.dart:35]: @hyungtaecf

**Comment by russellwheatley** [comment]
@hyungtaecf You will need to add an ignore line above your code change like this:

```dart
 FirebaseException({
    required this.plugin,
    this.message,
    String? code,
    this.stackTrace,
  // ignore: unnecessary_this
  }): this.code = code ?? 'unknown';
```

You need to install [melos tool](https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md#3-environment-setup). Run `melos run format` from root of project. Then CI will pass.

**Pull request merged**